test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Writer/JsonLight/ODataJsonLightInheritComplexCollectionWriterTests.cs

```diff
@@ -221,23 +226,83 @@ private static void WriteAndValidateSync(IEdmTypeReference itemTypeReference, OD
             ValidateWrittenPayload(stream, expectedPayload);
         }
 
-        //private static void WriteAndValidateAsync(IEdmTypeReference itemTypeReference, ODataResourceSet collectionStart, IEnumerable<ODataResource> items, string expectedPayload, bool writingResponse)
-        //{
-        //    MemoryStream stream = new MemoryStream();
-        //    var outputContext = CreateJsonLightOutputContext(stream, writingResponse, synchronous: false);
-        //    var createODataWriterTask = outputContext.CreateODataResourceSetWriterAsync(null, itemTypeReference == null ? null : itemTypeReference.ToStructuredType());
-        //    createODataWriterTask.Wait();
-        //    var odataWriter = createODataWriterTask.Result;
-        //    odataWriter.WriteStartAsync(collectionStart).Wait();
-        //    foreach (ODataResource item in items)
-        //    {
-        //        odataWriter.WriteStartAsync(item).Wait();
-        //        odataWriter.WriteEndAsync().Wait();
-        //    }
-
-        //    odataWriter.WriteEndAsync().Wait();
-        //    ValidateWrittenPayload(stream, expectedPayload);
-        //}
+        private static void WriteAndValidateAsync(IEdmTypeReference itemTypeReference, ODataResourceSet collectionStart, IEnumerable<ODataResource> items, string expectedPayload, bool writingResponse)
+        {
+            MemoryStream stream = new MemoryStream();
+            var outputContext = CreateJsonLightOutputContext(stream, writingResponse, synchronous: false);
+            var createODataWriterTask = outputContext.CreateODataResourceSetWriterAsync(null, itemTypeReference == null ? null : itemTypeReference.ToStructuredType());
+            createODataWriterTask.Wait();
+            var odataWriter = createODataWriterTask.Result;
+            odataWriter.WriteStartAsync(collectionStart).Wait();
+            foreach (ODataResource item in items)
+            {
+                odataWriter.WriteStartAsync(item).Wait();
+                odataWriter.WriteEndAsync().Wait();
+            }
+
+            odataWriter.WriteEndAsync().Wait();
+            ValidateWrittenPayload(stream, expectedPayload);
+        }
+
+        private static void WriteAndValidatePrimitivesSync(IEdmTypeReference itemTypeReference, ODataResourceSet collectionStart, IEnumerable<ODataResource> items, string expectedPayload, bool writingResponse)

```

Comment:
>collectionStart [](start = 113, length = 15)

resourceSet ?

Reply:
I just copied the pattern that was there before. The arguments that get passed in are named "collectionStartWithSerializationInfo" and "collectionStartWithoutSerializationInfo", which I think is why this parameter was originally named "collectionStart".

---
In reply to: [427635707](https://github.com/OData/odata.net/pull/1770#discussion_r427635707) [](ancestors = 427635707)